bokeh/charts/_builder.py

```diff
@@ -76,8 +81,26 @@ class Builder(HasProps):
 
     x_range = Instance(Range)
     y_range = Instance(Range)
+    y_names = Seq(String)
+    x_names = Seq(String)
 
     palette = Seq(Color, default=DEFAULT_PALETTE)
+    source = Instance(ColumnDataSource)
+
+    index = Any(help="""
+    An index to be used for all data series as follows:
+
+    - A 1d iterable of any sort that will be used as
+       series common index
+
+    - As a string that corresponds to the key of the
+       mapping to be used as index (and not as data
+       series) if area.values is a mapping (like a dict,
+       an OrderedDict or a pandas DataFrame)
+
+   """)
```

Comment:
I'll move this to the custom builders that actually use it. There's one feedback I'd like to have here which is actually if we should keep `index` and if the current interface is actually a good interface. Right know it can be used alternatively to x_names. So if you pass `df` and your index is not part of `df` and you don't want the default one you can specify your own iterable to by used with `index`. You could do the same just adding it as a field to `df` and specifying it's name in `x_name`. 

Opinions about this?